r2i/onnxrt/engine.h

```diff
@@ -0,0 +1,63 @@
+/* Copyright (C) 2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+*/
+
+#ifndef R2I_ONNXRT_ENGINE_H
+#define R2I_ONNXRT_ENGINE_H
+
+#include <r2i/iengine.h>
+
+#include <memory>
+#include <vector>
+
+#include <r2i/onnxrt/frame.h>
+#include <r2i/onnxrt/model.h>
+#include <r2i/onnxrt/prediction.h>
+
+namespace r2i {
+namespace onnxrt {
+
+class Engine : public IEngine {
+ public:
+  Engine ();
+  ~Engine ();
+
+  r2i::RuntimeError SetModel (std::shared_ptr<r2i::IModel> in_model) override;
+
+  r2i::RuntimeError Start () override;
+
+  r2i::RuntimeError Stop () override;
+
+  std::shared_ptr<r2i::IPrediction> Predict (std::shared_ptr<r2i::IFrame>
+      in_frame, r2i::RuntimeError &error) override;
+
+ protected:
```

Comment:
I think this should be private for now. Set it as protected is useful if there is an extension of this class (a child class) and allow that class to have access to these members.

Reply:
Agree, private members should be preferable to avoid issues.